src/electos/ballotmaker/demo_data/ballot_lab_data.py

```diff
@@ -84,19 +84,67 @@ def candidate_name(candidate: Candidate):
 
 
 def candidate_party(candidate: Candidate, index):
-    """Get the name and abbreviation of the party of a candidate as it appears on a ballot."""
-    party = index.by_id(candidate.party_id)
-    name = text_content(party.name) if party else ""
-    abbreviation = (
-        text_content(party.abbreviation)
-        if party and party.abbreviation
-        else ""
-    )
-    result = {
-        "name": name,
-        "abbreviation": abbreviation,
-    }
-    return result
+    """Get the name and abbreviation of the party of a candidate as it appears on a ballot.
+
+       Drop either field from result if it isn't present.
+    """
+    # Note: party ID is returned to allow de-duplicating parties in callers.
+    id_ = candidate.party_id
+    party = index.by_id(id_)
+    name = text_content(party.name) if party else None
+    abbreviation = text_content(party.abbreviation) if party and party.abbreviation else None
+    result = {}
+    if name:
+        result["name"] = name
+    if abbreviation:
+        result["abbreviation"] = abbreviation
+    return result, id_
+
+
+def candidate_contest_candidates(contest: CandidateContest, index):
```

Comment:
Although not mandatory, I do appreciate type hints for a function return value.

In this case, I think it's:

```
...)-> dict:
```

Reply:
Yeah, I'm being sloppy on the type hints. At this point I used them only to remind myself of the type.

I intend to go back and add type hints when things stabilize, and have `mypy` actually verify them.

Reply:
A couple of reasons why I think type hints for function & method returns are very helpful, right from the start:

* Having to decide what type of data your function returns helps me clarify exactly what my function is actually doing.
* If I can clarify my understanding of what the functions do, then it's much easier to write tests for them.
* Type hints help code reviewers understand your functions, too.

Of course, all of this is trivial to do when you write your tests first!